### PR TITLE
BE-590: hgraph: Refactor entity query handlers into `entity/query` submodule

### PR DESCRIPTION
**Pull request by indietyp**
## 🌟 What is the purpose of this PR?

The entity query request types and their associated HTTP handler functions have been refactored out of a single flat module into a dedicated `entity/query` submodule hierarchy. As part of this, the old `entity_query_request.rs` file — which contained a large amount of dead code related to a planned HashQL integration, proxy deserialization structs, and a four-variant subgraph enum — has been removed entirely.

The new `QueryEntitiesRequest` and `QueryEntitySubgraphRequest` types are simpler: `QueryEntitiesRequest` is now a plain struct with a `filter` field directly (no `EntityQuery` enum wrapping a filter or raw HashQL value), and `QueryEntitySubgraphRequest` is a two-variant untagged enum (`Paths` / `ResolveDepths`) rather than four variants. The `into_params` and `into_traversal_params` methods are now on the request types themselves rather than on a separate `EntityQueryOptions` struct, removing an intermediate layer.

The `Interactive` header parameter has also been removed from the `query_entities` and `query_entity_subgraph` OpenAPI path definitions, as it was only relevant to the HashQL compilation path that no longer exists in these handlers.

## 🔗 Related links

- https://linear.app/hash/issue/BE-537/hashql-remove-old-backend-wire-up-hashql-in-the-api

## 🔍 What does this change?

- Converts `entity.rs` into `entity/mod.rs` and extracts query handlers into `entity/query/mod.rs` and request types into `entity/query/request.rs`.
- Removes `entity_query_request.rs` and all dead code it contained: `EntityQuery`, `EntityQueryOptions`, `FlatQueryEntitiesRequestData`, `CompilationOptions`, HashQL diagnostic rendering, and the four-variant `QueryEntitySubgraphRequest`.
- Replaces the `EntityQuery::Filter` / `EntityQuery::Query` split with a direct `filter: Filter<'q, Entity>` field on `QueryEntitiesRequest`.
- Moves `into_params` and `into_traversal_params` onto `QueryEntitiesRequest` and `QueryEntitySubgraphRequest` respectively, eliminating `EntityQueryOptions` as a separate type.
- Removes the `Interactive` header from `query_entities` and `query_entity_subgraph` handler signatures and OpenAPI docs.
- Updates benchmark code in `tests/graph/benches` to use the simplified request types directly, removing the `into_parts` / `from_parts` round-trips that previously extracted an `EntityQuery` and `EntityQueryOptions`.
- Adds unit tests in `entity/query/request.rs` covering deserialization of both request types, including required-field validation and correct disambiguation of the `Paths` vs `ResolveDepths` subgraph variants.

## 🛡 What tests cover this?

- New unit tests in `libs/@local/graph/api/src/rest/entity/query/request.rs` covering minimal and fully-populated `QueryEntitiesRequest` deserialization, rejection of missing required fields, and correct parsing of both `QueryEntitySubgraphRequest` variants including edge cases around ontology vs entity traversal edges.

## ❓ How to test this?

1. Checkout the branch.
2. Run `cargo test` in `libs/@local/graph/api`.
3. Exercise the `/entities/query`, `/entities/query/subgraph`, and `/entities/query/count` endpoints and confirm they behave as before.

### CODE REVIEW

**Comment by indietyp** [comment]
* **#8895** <a href="https://app.graphite.com/github/pr/hashintel/hash/8895?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8894** <a href="https://app.graphite.com/github/pr/hashintel/hash/8894?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8887** <a href="https://app.graphite.com/github/pr/hashintel/hash/8887?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8882** <a href="https://app.graphite.com/github/pr/hashintel/hash/8882?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8870** <a href="https://app.graphite.com/github/pr/hashintel/hash/8870?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8853** <a href="https://app.graphite.com/github/pr/hashintel/hash/8853?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8827** <a href="https://app.graphite.com/github/pr/hashintel/hash/8827?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8841** <a href="https://app.graphite.com/github/pr/hashintel/hash/8841?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8840** <a href="https://app.graphite.com/github/pr/hashintel/hash/8840?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/hashintel/hash/8840?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#8839** <a href="https://app.graphite.com/github/pr/hashintel/hash/8839?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8838** <a href="https://app.graphite.com/github/pr/hashintel/hash/8838?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8837** <a href="https://app.graphite.com/github/pr/hashintel/hash/8837?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8836** <a href="https://app.graphite.com/github/pr/hashintel/hash/8836?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#8835** <a href="https://app.graphite.com/github/pr/hashintel/hash/8835?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: libs/@local/graph/api/src/rest/mod.rs [hunk 1]
```diff
@@ -16,7 +16,6 @@ pub mod admin;
 pub mod http_tracing_layer;
 pub mod jwt;
 
-mod entity_query_request;
 mod json;
 mod utoipa_typedef;
```
[the remark this answers is not in the document]
Comment (indietyp) [reply libs/@local/graph/api/src/rest/mod.rs:20]: used upstack

File: libs/@local/graph/api/src/rest/entity/query/request.rs [hunk 2]
```diff
@@ -0,0 +1,836 @@
...
+        #[serde(borrow, flatten)]
+        request: QueryEntitiesRequest<'q, 's, 'p>,
+    },
+    #[serde(rename_all = "camelCase")]
+    Paths {
+        traversal_paths: Vec<TraversalPath>,
+        #[serde(borrow, flatten)]
+        request: QueryEntitiesRequest<'q, 's, 'p>,
+    },
+}
```
[the remark this answers is not in the document]
Comment (indietyp) [reply libs/@local/graph/api/src/rest/entity/query/request.rs:240]: which is why we have a test for that

File: tests/graph/benches/manual_queries/entity_queries/mod.rs [hunk 3]
```diff
@@ -342,33 +336,19 @@ where
...
-            };
-
             let _response = store
                 .query_entities(
                     request.actor_id,
-                    options
-                        .into_params(filter, config)
-                        .expect("limit should not exceed configured maximum"),
+                    request.request.into_params_unchecked(config, None),
                 )
```
[the remark this answers is not in the document]
Comment (indietyp) [reply tests/graph/benches/manual_queries/entity_queries/mod.rs:321]: intentional to test the limits.

File: tests/graph/benches/manual_queries/entity_queries/mod.rs [hunk 4]
```diff
@@ -342,33 +336,19 @@ where
...
-            };
-
             let _response = store
                 .query_entity_subgraph(
                     request.actor_id,
-                    options
-                        .into_traversal_params(filter, traversal, config)
-                        .expect("limit should not exceed configured maximum"),
+                    request.request.into_traversal_params_unchecked(config),
                 )
```
[the remark this answers is not in the document]
Comment (indietyp) [reply tests/graph/benches/manual_queries/entity_queries/mod.rs:330]: intentional to test the limits.

File: libs/@local/graph/api/src/rest/entity/query/mod.rs [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+#[utoipa::path(
+    post,
+    path = "/entities/query",
+    request_body = QueryEntitiesRequest,
+    tag = "Entity",
+    params(
+        ("X-Authenticated-User-Actor-Id" = ActorEntityUuid, Header, description = "The ID of the actor which is used to authorize the request"),
+        ("after" = Option<String>, Query, description = "The cursor to start reading from"),
+        ("limit" = Option<usize>, Query, description = "The maximum number of entities to read"),
+    ),
```
[the remark this answers is not in the document]
Comment (indietyp) [reply libs/@local/graph/api/src/rest/entity/query/mod.rs:39]: same as before

**Review by TimDiekmann — APPROVED** [review]
